dali/image/tiff.cc

```diff
@@ -38,24 +37,16 @@ bool is_little_endian(const unsigned char *tiff) {
   return true;
 }
 
-}  // namespace
-
-TiffImage::TiffImage(const uint8_t *encoded_buffer, size_t length, dali::DALIImageType image_type) :
-        GenericImage(encoded_buffer, length, image_type) {
-}
-
-
-Image::ImageDims TiffImage::PeekDims(const uint8_t *encoded_buffer, size_t length) const {
-  DALI_ENFORCE(encoded_buffer);
-
+kernels::TensorShape<3> PeekDimsImpl(const uint8_t *encoded_buffer, size_t length) {
   TiffBuffer buffer(
-          std::string(reinterpret_cast<const char *>(encoded_buffer), static_cast<size_t>(length)),
-          is_little_endian(encoded_buffer));
+    std::string(reinterpret_cast<const char *>(encoded_buffer),
+    static_cast<size_t>(length)),
+    is_little_endian(encoded_buffer));
 
   const auto ifd_offset = buffer.Read<uint32_t>(4);
   const auto entry_count = buffer.Read<uint16_t>(ifd_offset);
   bool width_read = false, height_read = false;
-  size_t width, height;
+  size_t width = 0, height = 0;
```

Comment:
Just change the type here to int64_t and avoid the cast in return?

Reply:
Will take care of it in #1280